docs/reference/elasticsearch/rest-apis/highlighting-examples.md

```diff
@@ -0,0 +1,866 @@
+      }
+    }
+  },
+  "rescore": {
+    "window_size": 50,
+    "query": {
+      "rescore_query": {
+        "match_phrase": {
+          "comment": {
+            "query": "foo bar",
+            "slop": 1
+          }
+        }
+      },
+      "rescore_query_weight": 10
+    }
+  },
+  "_source": false,
+  "highlight": {
+    "order": "score",
+    "fields": {
+      "comment": {
+        "fragment_size": 150,
+        "number_of_fragments": 3,
+        "highlight_query": {
+          "bool": {
+            "must": {
+              "match": {
+                "comment": {
+                  "query": "foo bar"
+                }
+              }
+            },
+            "should": {
+              "match_phrase": {
+                "comment": {
+                  "query": "foo bar",
+                  "slop": 1,
+                  "boost": 10.0
+                }
+              }
+            },
+            "minimum_should_match": 0
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+
+## Set highlighter type [set-highlighter-type]
+
+The `type` field allows to force a specific highlighter type. The allowed values are: `unified`, `plain` and `fvh`. The following is an example that forces the use of the plain highlighter:
+
+```console
+GET /_search
+{
+  "query": {
+    "match": { "user.id": "kimchy" }
+  },
+  "highlight": {
+    "fields": {
+      "comment": { "type": "plain" }
+    }
+  }
+}
+```
+
+
+## Configure highlighting tags [configure-tags]
+
+By default, the highlighting will wrap highlighted text in `<em>` and `</em>`. This can be controlled by setting `pre_tags` and `post_tags`, for example:
+
+```console
+GET /_search
+{
+  "query" : {
+    "match": { "user.id": "kimchy" }
+  },
+  "highlight" : {
+    "pre_tags" : ["<tag1>"],
+    "post_tags" : ["</tag1>"],
+    "fields" : {
+      "body" : {}
+    }
+  }
+}
+```
+
+When using the fast vector highlighter, you can specify additional tags and the "importance" is ordered.
+
+```console
+GET /_search
+{
+  "query" : {
+    "match": { "user.id": "kimchy" }
+  },
+  "highlight" : {
+    "pre_tags" : ["<tag1>", "<tag2>"],
+    "post_tags" : ["</tag1>", "</tag2>"],
+    "fields" : {
+      "body" : {}
+    }
+  }
+}
+```
+
+You can also use the built-in `styled` tag schema:
+
+```console
+GET /_search
+{
+  "query" : {
+    "match": { "user.id": "kimchy" }
+  },
+  "highlight" : {
+    "tags_schema" : "styled",
+    "fields" : {
+      "comment" : {}
+    }
+  }
+}
+```
+
+
+## Highlight in all fields [highlight-all]
+
+By default, only fields that contains a query match are highlighted. Set `require_field_match` to `false` to highlight all fields.
+
+```console
+GET /_search
+{
+  "query" : {
+    "match": { "user.id": "kimchy" }
+  },
+  "highlight" : {
+    "require_field_match": false,
+    "fields": {
+      "body" : { "pre_tags" : ["<em>"], "post_tags" : ["</em>"] }
+    }
+  }
+}
+```
+
+
+## Combine matches on multiple fields [matched-fields]
+
+::::{warning}
```

Comment:
This doesn't feel like it should necessarily warrant a warning, should we downgrade this to note?